### ISSUE TITLE
syncing breaks when deleting auto-generated contact

### ISSUE THREAD

**Issue by tubeldelta** [Foundry376/Mailspring#1722]
##### Are there any related issues?

No. The problem occurs when deleting an auto-generated contact in the contacts window and it breaks syncing of the account until the cache is rebuilt, making it essentially impossible to delete any auto-generated contact.
...

##### What operating system are you using?

MacOS and Windows 10

...

##### What version of Mailspring are you using?
1.7.2

--

**Bug?**

##### Is this an email sync or connection issue?

Sync issue

See here: 

````
"aliases": [],
        "syncState": "sync_error",
        "syncError": {
          "code": 3,
          "error": {
            "stack": "Error: null\r\n *\r\n * An abort error occurred during program execution.\r\n * This error is thrown by system functions that detect corrupt state.\r\n *\r\nCREATE PROCESS FAIL: \r\naddr2line.exe -f -i -C -s -p -e \"c:\\users\\USERNAME\\appdata\\local\\mailspring\\app-1.7.2\\resources\\app.asar.unpacked\\mailsync\" 007DD7DF 007DBC63 762ED218 762EE6D2 762EDABB 6814625B 68146557 6814D8D6 77518DA2 77518D74 77504216 681476E6 006C0703 006B9CA0 006B00E6 00739331 007346BF 007277DD 006F81F7 006F85FE\r\n7784 [2019-11-28 22:04:02.808] [foreground] [critical]  * Stack trace:\r\n *   raise\r\n *   abort\r\n *   terminate\r\n *   _is_exception_typeof\r\n *   _is_exception_typeof\r\n *   RtlInterlockedCompareExchange64\r\n *   RtlInterlockedCompareExchange64\r\n *   KiUserExceptionDispatcher\r\n *   _CxxThrowException\r\n *\r\n\r\r\n\n    at ChildProcess.onStreamCloseOrExit (file:///C:/Users/appveyor/AppData/Local/Temp/1/nylas-build/electron-packager/win32-ia32/Mailspring-win32-ia32/resources/app/src/mailsync-process.ts:289:17)\n    at ChildProcess.emit (events.js:182:13)\n    at ChildProcess.EventEmitter.emit (domain.js:442:20)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)",
            "message": "null\r\n *\r\n * An abort error occurred during program execution.\r\n * This error is thrown by system functions that detect corrupt state.\r\n *\r\nCREATE PROCESS FAIL: \r\naddr2line.exe -f -i -C -s -p -e \"c:\\users\\USERNAME\\appdata\\local\\mailspring\\app-1.7.2\\resources\\app.asar.unpacked\\mailsync\" 007DD7DF 007DBC63 762ED218 762EE6D2 762EDABB 6814625B 68146557 6814D8D6 77518DA2 77518D74 77504216 681476E6 006C0703 006B9CA0 006B00E6 00739331 007346BF 007277DD 006F81F7 006F85FE\r\n7784 [2019-11-28 22:04:02.808] [foreground] [critical]  * Stack trace:\r\n *   raise\r\n *   abort\r\n *   terminate\r\n *   _is_exception_typeof\r\n *   _is_exception_typeof\r\n *   RtlInterlockedCompareExchange64\r\n *   RtlInterlockedCompareExchange64\r\n *   KiUserExceptionDispatcher\r\n *   _CxxThrowException\r\n *\r\n\r\r\n"
          },
          "signal": null
        },
        "__cls": "Account"
      }
    ],
    "accountsVersion": 11,
    "signatures": {
      "initial": {
        "id": "initial",
        "title": "Default",

````
...

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

Only tested with IMAP

...

##### Is the issue reproducible with a particular attachment, message, signature, etc?

No

...

--

**Feature Request?**

##### Does this feature exist in another mail client or tool you use?

...

**Comment by tubeldelta** [comment]
@bengotow I'm not sure if the mailsync log info is sufficient - I can easily reproduce this, so please let me know if you need more details.

**Comment by bengotow** [comment]
Hey! Sorry for the delay, I've been able to reproduce this and it will be fixed in the next release later today 👍

**Comment by bengotow** [comment]
Hey folks, thanks for reporting this. Happy to say this has been resolved in 1.7.6 which shipped this afternoon. 🚀  The app should auto-update, but if it doesn't you can also grab the new release from GitHub's Releases tab or from https://getmailspring.com/download. Thanks for your patience and help tracking this down, enjoy the update!
